agent/api/statechange.go

```diff
@@ -141,6 +148,13 @@ func NewContainerStateChangeEvent(task *apitask.Task, cont *apicontainer.Contain
 	return event, nil
 }
 
+// NewAttachmentStateChangeEvent creates a new attachment state change event
+func NewAttachmentStateChangeEvent(eniAttachment *apieni.ENIAttachment) AttachmentStateChange {
```

Comment:
because each state change (TaskStateChange, ContainerStateChange, AttachmentStateChange) implements the [Event](https://github.com/aws/amazon-ecs-agent/blob/master/agent/statechange/statechange.go#L31) interface and is treated as a event when it's sent to the event channel.. container state change and task state change are also constructed similarly in NewContainerStateChangeEvent and NewTaskStateChangeEvent